sympy/physics/continuum_mechanics/tests/test_beam.py

```diff
@@ -598,3 +599,55 @@ def test_parabolic_loads():
     loading = beam.load.xreplace({L: 10, E: 20, I: 30, P: 40})
     assert loading.xreplace({x: 5}) == 40 * 5**8
     assert loading.xreplace({x: 15}) == 0
+
+
+def test_cross_section():
+    l = Symbol('l')
+    E = Symbol('E')
+    C3, C4 = symbols('C3, C4')
+    a, c, g, h, r = symbols('a, c, g, h, r')
+
+    # beam with a circular cross-section
+    b1 = Beam(50, E, Circle((0, 0), r))
+    assert b1.cross_section == Circle((0, 0), r)
+    assert b1.second_moment == pi*r*Abs(r)**3/4
```

Comment:
Are the lines above needed? I thought the whole point was that we can pass a shape argument when constructing the beam and then the second moment would be automatically calculated.

Also is it necessary in the API to be able to modify the cross_section after creating the beam (seems like poor API to me)? If it is necessary then at least it should be enough to set the cross_section and have the second_moment be calculated implicitly from that.

Reply:
It is being calculated implicitly, these two lines are just testing whether correct values has been calculated (using assert).

Reply:
But I think it won't be able to update it if the cross_section is explicitly changed later. We should better make `cross_section` read-only.

Reply:
Sorry I misread the lines above as changing the properties not checking them. That seems fine.

In general I don't see the benefit in being able to change the geometry after the object is created but beam seems to have a bunch of things like this (`add_load` etc).

Reply:
> Also is it necessary in the API to be able to modify the cross_section after creating the beam (seems like poor API to me)? 

I'd be fine with making cross section "private" attribute, then we don't need to support people changing it after creating a beam. You can also not have a setter, to make it impossible to modify.

The Beam class is a mutable object for better or worse. It lets you construct it piecemeal instead of all in one go. This is likely an advantage for complex beams and for interactive work. Maybe we should stick with following its existing design patterns.

Reply:
Should we make cross-section read-only instead of private, by removing setter but retaining property? 
Also does the `if statement` has to be shifted to `__init__` or is it fine there? I think with the setter being removed it should have no problem with the Beam3D class.

Reply:
Yeah, we are making `_cross_section` read-only. You have to remove the `cross_section` setter for that. We don't want the user to change `_cross_section` attribute using it. As we are making it read-only, I think property method should be there.